### PR TITLE
fix: AU-1760, AU-1761, AU-1763, AU-1773, AU-1794: liikunta & tilankaytto feedback fixes

### PR DESCRIPTION
**Pull request by suvikankare**
## What was done

* [AU-1760](https://helsinkisolutionoffice.atlassian.net/browse/AU-1760) Aktiiviharrastajat as required
* [AU-1761](https://helsinkisolutionoffice.atlassian.net/browse/AU-1761) Update attachment text
* [AU-1763](https://helsinkisolutionoffice.atlassian.net/browse/AU-1763) 
  1. Re-style Lajijaostot
  2. Laji required 
  3. At least one of the age groups required
  4. If age group field has value, practice hours required
  5. Reordered fields to men - women - others
* [AU-1773](https://helsinkisolutionoffice.atlassian.net/browse/AU-1773) Hide fieldset Jäsenet
* [AU-1794](https://helsinkisolutionoffice.atlassian.net/browse/AU-1794) Change term 'Subsidy for use of space' to 'Facility usage grant'

## How to install

* Make sure your instance is up and running on correct branch.
  * `git checkout feature/AU-1760-liikunta-tilankaytto`
  * `make fresh`
* Run `make drush-cr`

## How to test

* [ ] [Open application](https://hel-fi-drupal-grant-applications.docker.so/fi/uusi-hakemus/liikunta_toiminta_ja_tilankaytto)
* [ ] Go through "What was done" and check that everything works

[AU-1760]: https://helsinkisolutionoffice.atlassian.net/browse/AU-1760?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[AU-1761]: https://helsinkisolutionoffice.atlassian.net/browse/AU-1761?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[AU-1763]: https://helsinkisolutionoffice.atlassian.net/browse/AU-1763?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[AU-1773]: https://helsinkisolutionoffice.atlassian.net/browse/AU-1773?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[AU-1794]: https://helsinkisolutionoffice.atlassian.net/browse/AU-1794?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by jiisuominen** [comment]
> SonarCloud Quality Gate failed.    [Quality Gate failed](https://sonarcloud.io/dashboard?id=City-of-Helsinki_avustusasiointi&pullRequest=892)
> 
> [Bug](https://sonarcloud.io/project/issues?id=City-of-Helsinki_avustusasiointi&pullRequest=892&resolved=false&types=BUG) [A](https://sonarcloud.io/project/issues?id=City-of-Helsinki_avustusasiointi&pullRequest=892&resolved=false&types=BUG) [0 Bugs](https://sonarcloud.io/project/issues?id=City-of-Helsinki_avustusasiointi&pullRequest=892&resolved=false&types=BUG) [Vulnerability](https://sonarcloud.io/project/issues?id=City-of-Helsinki_avustusasiointi&pullRequest=892&resolved=false&types=VULNERABILITY) [A](https://sonarcloud.io/project/issues?id=City-of-Helsinki_avustusasiointi&pullRequest=892&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=City-of-Helsinki_avustusasiointi&pullRequest=892&resolved=false&types=VULNERABILITY) [Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=City-of-Helsinki_avustusasiointi&pullRequest=892&resolved=false&types=SECURITY_HOTSPOT) [A](https://sonarcloud.io/project/security_hotspots?id=City-of-Helsinki_avustusasiointi&pullRequest=892&resolved=false&types=SECURITY_HOTSPOT) [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=City-of-Helsinki_avustusasiointi&pullRequest=892&resolved=false&types=SECURITY_HOTSPOT) [Code Smell](https://sonarcloud.io/project/issues?id=City-of-Helsinki_avustusasiointi&pullRequest=892&resolved=false&types=CODE_SMELL) [A](https://sonarcloud.io/project/issues?id=City-of-Helsinki_avustusasiointi&pullRequest=892&resolved=false&types=CODE_SMELL) [4 Code Smells](https://sonarcloud.io/project/issues?id=City-of-Helsinki_avustusasiointi&pullRequest=892&resolved=false&types=CODE_SMELL)
> 
> [No Coverage information](https://sonarcloud.io/component_measures?id=City-of-Helsinki_avustusasiointi&pullRequest=892) No Coverage information [](https://sonarcloud.io/component_measures?id=City-of-Helsinki_avustusasiointi&pullRequest=892&metric=new_duplicated_lines_density&view=list) [19.0% Duplication](https://sonarcloud.io/component_measures?id=City-of-Helsinki_avustusasiointi&pullRequest=892&metric=new_duplicated_lines_density&view=list)
> 
> idea Catch issues before they fail your Quality Gate with our IDE extension sonarlint [SonarLint](https://www.sonarsource.com/products/sonarlint/features/connected-mode/?referrer=sonarcloud-welcome)

Oon nostanu ton koodikopioinnin -> 30% tän tsekin jälkeen. Uus kommitti varmasti ratkasee sen.

**Comment by oskarimysteerio** [comment]
Olikohan "Myös yksi ikäluokka on pakollinen.", että pitää olla joko Senioireita / Adult / Junior vai että jokaisessa groupissa pitää olla 1 arvo 🤔

File: conf/cmi/webform.webform.liikunta_toiminta_ja_tilankaytto.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ third_party_settings:
       70: '70'
     applicationTargetGroup: '26'
     applicationOpen: '2022-10-03T11:08:26'
-    applicationClose: '2023-08-31T11:19:00'
+    applicationClose: '2024-02-29T11:19:00'
```
Comment (oskarimysteerio) [inline conf/cmi/webform.webform.liikunta_toiminta_ja_tilankaytto.yml:22]: Tää vissiin valunu lokaalin yliajoista?

File: public/modules/custom/grants_club_section/src/Validator/FieldValueValidator.php [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+   *
+   * Note that #required is validated by _form_validate() already.
+   */
+  public static function validateSeniorHours(&$element, FormStateInterface $form_state) {
+    $parents = $element['#parents'];
+    $field = array_pop($parents);
+    $parent = $form_state->getValue($parents);
+    $tOpts = ['context' => 'grants_club_section'];
+
+    if (!empty($parent['senioMen']) || !empty($parent['seniorWomen']) || !empty($parent['seniorOthers'])) {
```
Comment (oskarimysteerio) [inline public/modules/custom/grants_club_section/src/Validator/FieldValueValidator.php:79]: Täst puuttu r
```suggestion
    if (!empty($parent['seniorMen']) || !empty($parent['seniorWomen']) || !empty($parent['seniorOthers'])) {
```

File: public/modules/custom/grants_club_section/src/Validator/FieldValueValidator.php [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+   *
+   * Note that #required is validated by _form_validate() already.
+   */
+  public static function validateAdultHours(&$element, FormStateInterface $form_state) {
+    $parents = $element['#parents'];
+    $field = array_pop($parents);
+    $parent = $form_state->getValue($parents);
+    $tOpts = ['context' => 'grants_club_section'];
+
+    if (!empty($parent['men']) || !empty($parent['women']) || !empty($parent['adultOthers'])) {
```
Comment (oskarimysteerio) [inline public/modules/custom/grants_club_section/src/Validator/FieldValueValidator.php:55]: Eiks tää oo vähän hassusti, jos joku ei oo tyhjä, niin lyödään virhe 😅 ? 

image

**Review by oskarimysteerio — CHANGES_REQUESTED** [review]

File: public/modules/custom/grants_club_section/src/Validator/FieldValueValidator.php [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+   *
+   * Note that #required is validated by _form_validate() already.
+   */
+  public static function validateAdultHours(&$element, FormStateInterface $form_state) {
+    $parents = $element['#parents'];
+    $field = array_pop($parents);
+    $parent = $form_state->getValue($parents);
+    $tOpts = ['context' => 'grants_club_section'];
+
+    $sfmnf = $element;
```
Comment (oskarimysteerio) [inline public/modules/custom/grants_club_section/src/Validator/FieldValueValidator.php:55]:
```suggestion
```

**Comment by oskarimysteerio** [comment]
Niin joo - falskaa viel nää Adult ja Junior validaatiot viel - my bad kun en merkannu jokasen kohalla tota virhettä

image

**Comment by oskarimysteerio** [comment]
Varmaan jossain vaihees tulee sanomista, mutta pitäiskö tällä preview sivullea ottaa jotain kantaa, kun noi jäsenet fieldset piilotettiin, nyt sinne tulee "haamuna" noi piilotetut?

image

**Review by oskarimysteerio — DISMISSED** [review]
Joo, eiköhän tässä oo punnerrettu tää nyt timanttiseks 😅

**Review by oskarimysteerio — APPROVED** [review]
Tai siis nyt.

**Pull request merged**